### PR TITLE
Soft Delete Payment, Receipt, and Invoice

### PR DESCRIPTION
**Pull request by rhimmelbauer**
Notes:
- Created SoftDeleteBase Model to add the deleted field and override the delete function in Models that implement it.
- Updated get_receipt to use created for both receipt and payment instead of the receipt.start_date
- Added admin actions to soft delete payments with no corresponding receipts or with invoices with no order_items that have been completed.
- Added Views to show Payments with no Receipts and with no Order Items.
- Update  japanese translations

<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

File: vendor/views/vendor_admin.py [hunk 1]
```diff
@@ -358,3 +359,35 @@ def post(self, request, *args, **kwargs):
...
         messages.info(request, _("Subscription Renewed"))
         return redirect(request.META.get('HTTP_REFERER', self.success_url))
+
+
+class PaymentWithNoReceiptListView(LoginRequiredMixin, ListView):
+    template_name = "vendor/manage/payment_list.html"
+    model = Payment
+
+    def get_queryset(self):
+        if hasattr(self.request, 'site'):
```
Comment (mgcamisa) [inline vendor/views/vendor_admin.py:369]: do you need this in here if you're setting the site from the request in line 371?

File: vendor/views/vendor_admin.py [hunk 2]
```diff
@@ -358,3 +359,35 @@ def post(self, request, *args, **kwargs):
...
+        context['title'] = _("Payments with no Receipts")
+        return context
+
+
+class PaymentWithNoOrderItemsListView(LoginRequiredMixin, ListView):
+    template_name = "vendor/manage/payment_list.html"
+    model = Payment
+
+    def get_queryset(self):
+        if hasattr(self.request, 'site'):
```
Comment (mgcamisa) [inline vendor/views/vendor_admin.py:385]: same here

File: vendor/views/vendor_admin.py (hunk 1 above)
Comment (rhimmelbauer) [reply vendor/views/vendor_admin.py]: yeah removed that unnecessary check if the function already does it

File: vendor/views/vendor_admin.py (hunk 2 above)
Comment (rhimmelbauer) [reply vendor/views/vendor_admin.py]: yeah removed that unnecessary check if the function already does it

**Review by mgcamisa — APPROVED** [review]

**Pull request merged**